### PR TITLE
chore(website): update config editor and add tabs to playground

### PR DESCRIPTION
**Pull request by armano2**
## PR Checklist

-   [x] Addresses an existing open issue: #5016
-   [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
-   [x] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md) were taken

## Overview

- configs are displayed as tabs
- reduce computation complexity when updating codeFixers
- <details><summary>In case of fatal parsing error fallback to error list, this unifies how errors are displayed between screens</summary>

  Old:
  image

  New: 
  image

  </details>

- ensure that all proper libs are download for specified typescript versions
- <details><summary>typescript errors markers are going to be now printed only for user provided file with exclusion to externally provided files</summary>

  Old:
  image

  New:
  image
  </details>

- <details><summary>small adjustments to mobile version</summary>

    image

  </details>
  
- any config provided by user will be saved but may not be used if validation/parsing fails for it
- we are currently using only compilerOptions and rules field from configs provided by user
- modal with visual config has been moved to right side after selecting tab
   image
- autocompletion for eslintrc and tsconfig has been added
   image
- json parsing errors are displayed live from monaco
   image
- url format has changed `tsconfig` and `eslintrc` are now used instead of `rules` and `tsConfig`, backward compatibility layer is in place 
- correct bug with event missing in monaco for ts <= 4.0.5 https://github.com/microsoft/monaco-editor/issues/313

### CODE REVIEW

**Comment by armano2** [comment]
for now I'd like to ask for your feedback about UI/UX of those changes, as code still requires some polishing,

File: packages/website/src/components/Playground.module.css [hunk 1]
```diff
@@ -37,3 +42,67 @@
...
+
+.tabCode {
+  height: calc(100%);
+}
+
+.tabStyle {
+  border: none;
+  border-right: 1px solid var(--playground-secondary-color);
+  background: var(--playground-main-color);
+  color: var(--ifm-color-emphasis-600);
```
Comment (JoshuaKGoldberg) [inline packages/website/src/components/Playground.module.css:61]: #8D949E doesn't have enough color contrast to be accessible against a white background. #606770 does, though. https://webaim.org/resources/contrastchecker/ 

```suggestion
  color: var(--ifm-color-emphasis-700);
```

File: packages/website/src/components/ASTViewerScope.tsx [hunk 2]
```diff
@@ -17,12 +17,8 @@ export default function ASTViewerScope({
...
   useEffect(() => {
-    if (typeof value === 'string') {
-      setModel(value);
-    } else {
-      const scopeSerializer = createScopeSerializer();
-      setModel(serialize(value, scopeSerializer));
-    }
+    const scopeSerializer = createScopeSerializer();
+    setModel(serialize(value, scopeSerializer));
   }, [value]);
```
Comment (JoshuaKGoldberg) [inline packages/website/src/components/ASTViewerScope.tsx:22]: We can go even simpler here and in the other two `ASTViewer(.+)` components:

```tsx
const astSerializer = createESTreeSerializer();

export default function ASTViewerESTree({
  value,
  position,
  onSelectNode,
}: ASTESTreeViewerProps): JSX.Element {
  const model = serialize(value, astSerializer);

  return (
    <ASTViewer value={model} position={position} onSelectNode={onSelectNode} />
  );
}
```

This is similar to the other thread about `useEffect` _(I'm forgetting where that was...)_: it's not generally needed to synchronize some data as the result of some inputs. You can use a variable directly instead, and the computed data (`model`) will reflect whatever the latest prop data (`value`) is at each render.

If performance is a concern you can use a `useMemo`:

```ts
const model = useMemo(() => serialize(value, astSerializer), [value]);
```

File: packages/website/src/components/EditorTabs.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  readonly change: (tab: TabType) => void;
+  readonly showModal: () => void;
+}
+
+export default function EditorTabs({
+  tabs,
+  activeTab,
+  change,
+  showModal,
+}: FileTabsProps): JSX.Element {
```
Comment (JoshuaKGoldberg) [inline packages/website/src/components/EditorTabs.tsx:18]: A few missing tab attributes around this component for accessibility...

* The div holding the buttons should have `role="tablist"`
* The buttons themselves should have `role="tab"`, `aria-selected`, and `aria-controls` props
* The openable components should have `role="tabpanel"`

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tab_role

File: packages/website/src/components/config/ConfigEslint.tsx [hunk 4]
```diff
@@ -1,14 +1,20 @@
...
+import type {
+  RulesRecord,
+  RuleDetails,
+  RuleEntry,
+  ConfigModel,
+} from '../types';
 import { shallowEqual } from '../lib/shallowEqual';
+import { parseESLintRC, toJsonConfig } from '@site/src/components/config/utils';
 
 export interface ModalEslintProps {
```
Comment (JoshuaKGoldberg) [inline packages/website/src/components/config/ConfigEslint.tsx:13]: Nit:

```suggestion
export interface ConfigEslintProps {
```

File: packages/website/src/components/config/ConfigTypeScript.tsx [hunk 5]
```diff
@@ -1,20 +1,14 @@
 import React, { useCallback, useEffect, useState } from 'react';
 
 import ConfigEditor, { ConfigOptionsType } from './ConfigEditor';
-import type { CompilerFlags } from '../types';
+import type { CompilerFlags, ConfigModel } from '../types';
 import { shallowEqual } from '../lib/shallowEqual';
+import { getTypescriptOptions, parseTSConfig, toJsonConfig } from './utils';
 
 interface ModalTypeScriptProps {
```
Comment (JoshuaKGoldberg) [inline packages/website/src/components/config/ConfigTypeScript.tsx:8]: Nit:

```suggestion
interface ConfigTypeScriptProps {
```

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Looks great! Feels better than always using a modal.

Requesting changes for the accessibility of the buttons. Long term we probably will want to set up a design system using some accessible base system to make tab components and the like easier. But for now this is a great start.

File: packages/website/src/components/EditorTabs.tsx (hunk 3 above)
Comment (armano2) [reply packages/website/src/components/EditorTabs.tsx:18]: `aria-controls` and `role="tabpanel"` doesn't apply in here, we actually have single content that is changes what is displayed

Comment (JoshuaKGoldberg) [reply packages/website/src/components/EditorTabs.tsx:18]: Hmm, I don't think I follow where you're coming from. It's a set of buttons, each of which have an onclick that changes the content in the page. That's a pretty canonical use for tabs, no?

From https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tab_role#frame_example:

Screenshot of the three tabbed frame example from the MDN page

Comment (armano2) [reply packages/website/src/components/EditorTabs.tsx:18]: ok, i added role tab, tablist and `aria-selected` let me know if that's correct now

Comment (JoshuaKGoldberg) [reply packages/website/src/components/EditorTabs.tsx:18]: Super, thanks!

I think there's more work that _could_ be done around `aria-controls`, but I'm not super familiar with this myself and don't want to block now that the basics are done.

**Review by JoshuaKGoldberg — APPROVED** [review]
:shipit: !

**Pull request merged**